python/nelsie/__main__.py

```diff
@@ -1,40 +1,73 @@
 import argparse
 import importlib
+import sys
 import traceback
+from typing import List, Set
+
 from . import nelsie as nelsie_rs
 from . import watch
 
 
 def parse_args():
     parser = argparse.ArgumentParser()
     subparser = parser.add_subparsers()
+    subparser.required = True
     watch_parser = subparser.add_parser("watch")
-    watch_parser.add_argument("source_filename")
+    watch_parser.add_argument("source_filename", nargs="+")
     return parser.parse_args()
 
 
-def load_py_file(filename):
+def get_system_modules() -> Set[str]:
+    from sys import modules
+    import importlib  # noqa
+
+    return set(modules.keys())
+
+
+def reload_and_get_watched_files(filenames: List[str], system_modules: Set[str]) -> List[str]:
+    """
+    Executes the first passed Python script from the `filenames` list
+    and gathers all files that should be watched.
+    """
     watch._WATCH_SET = set()
-    importlib.machinery.SourceFileLoader("slides", filename).load_module()
+
+    # Try to reload non-system modules, to refresh Python imports from the main file
+    # We need to skip some system modules, otherwise reloading them could break Python
+    for name, module in dict(sys.modules).items():
+        if name in sys.builtin_module_names or name in system_modules or name.startswith("nelsie."):
+            continue
+        try:
+            importlib.reload(module)
+        except BaseException:
+            pass
+
+    # Re-execute the file
+    importlib.machinery.SourceFileLoader("slides", filenames[0]).load_module()
+
     s = watch._WATCH_SET
     watch._WATCH_SET = None
-    s.add(filename)
-    return list(s)
+
+    to_watch = list(filenames)
+    to_watch.extend(list(s))
```

Comment:
This is not correct. Watched images will grow forever with duplicates everytime the slides are rebuild.

Reply:
The second call should have just passed `source_filenames`, not `files`, thanks.

Reply:
btw: you can just write `.extend(x)` not `.extend(list(x))`